circus-train-aws/src/main/java/com/hotels/bdp/circustrain/aws/AssumeRoleCredentialProvider.java

```diff
@@ -31,30 +31,35 @@
   public static final String ASSUME_ROLE_PROPERTY_NAME = "com.hotels.bdp.circustrain.aws.AssumeRoleCredentialProvider.assumeRole";
   private static final int CREDENTIALS_DURATION = 12 * 60 * 60; // max duration in seconds for assumed role credentials
 
-  private AWSCredentials credentials;
   private final Configuration conf;
+  private STSAssumeRoleSessionCredentialsProvider credProvider;
 
   public AssumeRoleCredentialProvider(Configuration conf) {
     this.conf = conf;
+    initializeCredProvider();
```

Comment:
move this to `getCredentials` and have it like:
```
if (this.credProvider == null) {
  initializeCredProvider();
}
```
This makes sure we throw the error in `getCredentials` instead of the constructor.  If we throw in the constructor, the provider chain will just silently construct the next provider in the chain.

Reply:
ah okay. Didn't understand the full context of that. Moving it now.